### ISSUE TITLE
RepositoryFileApi's getOptionalFileInfo method can return false Optional.empty()

### ISSUE THREAD

**Issue by amitt00r** [gitlab4j/gitlab4j-api#1095]
We recently faced one issue where below method returned false Optional.empty() for a file which did exist. On checking the code found issue with below code in RepositoryFileApi class.
  
```
  public Optional<RepositoryFile> getOptionalFileInfo(Object projectIdOrPath, String filePath, String ref) {
        try {
            return Optional.ofNullable(this.getFileInfo(projectIdOrPath, filePath, ref));
        } catch (GitLabApiException var5) {
            return GitLabApi.createOptionalFromException(var5);
        }
    }
```

Issue:
Method above doesn't do status code checks on GitLabApiException before returning Optional.empty(). 

Expected Behaviour:
This should first check on 'httpStatus' == 404 , and then return optional.empty(). All other exceptions should be rethrown.

**Comment by jmini** [comment]
Interesting finding.

The problem I see with "rethrow" is that all the `getOptionalXXX()` is that they are not throwing any Exception.

**Comment by amitt00r** [comment]
Yep, I found it later all methods returning Optional have this issue, all use same method `GitlabApi.createOptionalFromException()` which never throws exception no matter what the status code is.
